theurgy.lic

```diff
@@ -1,186 +1,449 @@
+    end
+  end
+  
+  def rituals_remain?
+    !@rituals.empty?
   end
 
-  ######################## COMMUNES ########################
+  def on_hand?(item)
+    if item[:name] == 'oil'
+      inside_container?('holy oil') || inside_container?('some oil')
+    elsif item[:any_container]
+      exists?(item[:name])
+    else
+      inside_container?(item[:name])
+    end
+  end
 
-  def comm_eluned
-    fput "stow #{GameObj.right_hand.noun}" if GameObj.right_hand.name != 'Empty'
-    fput "stow #{GameObj.left_hand.noun}" if GameObj.left_hand.name != 'Empty'
+  def inside_container?(item_name)
+    inside?(item_name, @theurgy_supply_container)
+  end
 
-    forage? 'dirt'
+  def put_in_container(item_name)
+    bput("put #{item_name} in my #{@theurgy_supply_container}",
+         'You put', 'What were you referring to')
+  end
 
-    get_from_container(@water_holder)
-    bput('commune eluned', 'completed this commune too recently', 'You grind some dirt in your fist')
-    put_in_container(@water_holder)
-    fput('drop dirt') if [left_hand, right_hand].include?('dirt')
+  def get_from_container(item_name)
+    bput("get #{item_name} from my #{@theurgy_supply_container}",
+         'You get', 'I could not find', 'What were you referring to')
+  end
+    
+  def have_holy_water?
+    inside?('holy water', @water_holder)    
   end
 
-  def comm_tamsine
-    if inside?('holy water', @water_holder)
-      pause 1
-      waitrt?
-      get_from_container(@water_holder)
-      fput "sprinkle #{@water_holder} on #{checkname}"
-      bput('commune tamsine', 'completed this commune too recently', 'You feel warmth spread throughout your body')
-      put_in_container(@water_holder)
-    else
-      Flags.delete('theurgy-tamsine')
-      nil
+  def buy_supplies
+    items_to_buy = (@rituals + @communes).map { |rc| rc[:items] }.flatten
+    items_to_buy.uniq! { |item| item[:name] }
+    items_to_buy.reject! { |item| item[:on_hand] }
+    items_to_buy.sort_by { |item| item[:shop]['id'] }
+    
+    ensure_copper_on_hand(600, @hometown) unless items_to_buy.empty?
+
+    items_to_buy.each do |item|
+      walk_to(item[:shop]['id'])
+      
+      if item[:purchase_method]
+        send(item[:purchase_method])
+      else
+        buy_item(item[:shop]['id'], item[:name])
+      end
+      put_in_container(item[:name])
     end
   end
 
-  def comm_kert
-    @hold_research = nil
-    if inside?('holy oil', @theurgy_supply_container)
-      get_from_container('holy oil')
-    else
-      case bput("get oil from #{@theurgy_supply_container}", 'holy oil', 'some oil', 'I could not find', 'What were you referring to')
-      when 'I could not find', 'What were you referring to'
-        return
-      when 'some oil'
-        research_check
-        fput 'prep bless'
-        pause 5
-        bput('cast oil', 'leaving it somehow changed', 'nothing else seems to happen', 'backfires')
+  def buy_taffelberries
+    bput('order 14',
```

Comment:
There's an `order_item_ helper in `common-items`

Reply:
It looks like for Taffelberries the messaging doesn't quite match the messaging in order_items.  I can't be sure for the parchment order since the original `pbut` in `theurgy.lic` doesn't completely overlap.  I can check when I'm next in Crossing but for now I'd like to leave this as-is since I know it works.